### PR TITLE
chore: configure GitHub template chooser

### LINKED ISSUE #431: Prevent empty issues, redirect issues that should go upstream
Once we're done with #430, we can think about forcing contributors to use the provided forms.

We can also configure the template chooser (screen you see when you click "new issue") to show a button with a link to upstream projects like ESLint with a custom text. This can help re-direct unwanted stuff to the proper place. 😉 

Example `config.yml`:

```yml
blank_issues_enabled: false
contact_links:
  - name: ESLint upstream
    url: https://github.com/eslint/eslint/issues
    about: Report problems with the ESLint program itself to the upstream project.
```

Read [GitHub Docs, configuring the template chooser](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser) to learn more.

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).
- [ ] If some rule is added/updated/removed, I've regenerated the rules list.
- [ ] If some rule meta info is changed, I've regenerated the plugin shared configs.

## Changes

- Block empty issues
- Redirect issues that should go to upstream ESLint project
- Add link to our discussions page on GitHub

## Context

Fixes #431.

**EDIT:** Now that I have created the PRs for the Issue Forms, it makes more sense to wait until those 3 Issue Form PRs have been merged into `main`, and then review and merge this PR.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/config.yml:5]: Feel free to suggest changes to `name`, `url` and `about` text. 😉

File: .github/ISSUE_TEMPLATE/config.yml
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/config.yml:1]: If you want to prevent empty issues, you'd set `blank_issues_enabled` to `false`, but that's something to do later. 😉

**Review by HonkingGoose — COMMENTED** [review]
General comments to help you understand what I'm doing here.

File: .github/ISSUE_TEMPLATE/config.yml
Comment (HonkingGoose) [reply .github/ISSUE_TEMPLATE/config.yml:1]:
```suggestion
blank_issues_enabled: false
```

File: .github/ISSUE_TEMPLATE/config.yml
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Wait until the Issue Form PRs have been merged into `main`:

- [ ] https://github.com/testing-library/eslint-plugin-testing-library/pull/433
- [ ] https://github.com/testing-library/eslint-plugin-testing-library/pull/434
- [ ] https://github.com/testing-library/eslint-plugin-testing-library/pull/435

File: .github/ISSUE_TEMPLATE/config.yml
Comment (Belco90) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Not really sure what's the idea with these links.

Comment (HonkingGoose) [reply .github/ISSUE_TEMPLATE/config.yml:5]: The idea is that you'll have a button on this screen: https://github.com/testing-library/eslint-plugin-testing-library/issues/new/choose with a link to the ESLint repository.

For an example of how those buttons look go here: https://github.com/progit/progit2/issues/new/choose

Comment (Belco90) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Aaah ok. I think this one is fine then. Should we add a link to our Discussions then?

- name: Ask a Question, Discuss
- URL: https://github.com/testing-library/eslint-plugin-testing-library/discussions
- about: Look for `eslint-plugin-testing-library` Questions and Discussions or start a new one

Comment (HonkingGoose) [reply .github/ISSUE_TEMPLATE/config.yml:5]:
```suggestion
  - name: Ask a Question, Discuss
    url: https://github.com/testing-library/eslint-plugin-testing-library/discussions
    about: Look for `eslint-plugin-testing-library` Questions and Discussions or start a new one
```

I'm not sure if we can use `monospaced text` in the `about` section... That part might not work properly. If things break we can just use normal plain text in the field. 😄

Comment (Belco90) [reply .github/ISSUE_TEMPLATE/config.yml]: Oh sorry, I didn't mean to replace the ESLint one but having both!

Comment (HonkingGoose) [reply .github/ISSUE_TEMPLATE/config.yml]: Whoops! 😄 I'll go add the ESLint one back as well. 😉

**Review by Belco90 — APPROVED** [review]

**Pull request merged**
